File: src/Addins/Eto.Forms.Templates/content/File-CSharp/MainForm.cs
#if (UseCode || UseCodePreview)
using System;
using Eto.Forms;
using Eto.Drawing;

namespace EtoApp._1
{
	public partial class MainForm : Form
	{
		public MainForm()
		{
#if (UseCodePreview)
			InitializeComponent();
#else
#if IsWindow
			Title = "My Eto Form";
			ClientSize = new Size(400, 350);
#endif

			Content = new StackLayout
			{
				Padding = 10,
				Items =
				{
					"Hello World!",
					// add more controls here
				}
			};

#if IsForm
			// create a few commands that can be used for the menu and toolbar
			var clickMe = new Command { MenuText = "Click Me!", ToolBarText = "Click Me!" };
			clickMe.Executed += (sender, e) => MessageBox.Show(this, "I was clicked!");

			var quitCommand = new Command { MenuText = "Quit", Shortcut = Application.Instance.CommonModifier | Keys.Q };
			quitCommand.Executed += (sender, e) => Application.Instance.Quit();

			var aboutCommand = new Command { MenuText = "About..." };
			aboutCommand.Executed += (sender, e) => new AboutDialog().ShowDialog(this);

			// create menu
			Menu = new MenuBar
			{
				Items =
				{
					// File submenu
					new ButtonMenuItem { Text = "&File", Items = { clickMe } },
					// new ButtonMenuItem { Text = "&Edit", Items = { /* commands/items */ } },
					// new ButtonMenuItem { Text = "&View", Items = { /* commands/items */ } },
				},
				ApplicationItems =
				{
					// application (OS X) or file menu (others)
					new ButtonMenuItem { Text = "&Preferences..." },
				},
				QuitItem = quitCommand,
				AboutItem = aboutCommand
			};

			// create toolbar			
			ToolBar = new ToolBar { Items = { clickMe } };
#endif
#endif
		}
	}
}
#endif

File: src/Eto.Forms.Templates/content/App-VisualBasic/EtoApp.1/MainForm.vb
#If UseCode Or UseCodePreview
Imports Eto.Forms
Imports Eto.Drawing

Public Partial Class MainForm
	Inherits Form

	Public Sub New()
#If UseCodePreview
		InitializeComponent()
#Else
#If IsWindow
		Title = "My Eto Form"
		MinimumSize = New Size(200, 200)
#End If
		Padding = 10

		Content = New StackLayout(
			"Hello World!" ' Add more controls here
		)

#If IsForm
		' create a few commands that can be used for the menu and toolbar
		Dim clickMe As New Command With { .MenuText = "Click Me!", .ToolbarText = "Click Me!" }
		AddHandler clickMe.Executed, Sub() MessageBox.Show(Me, "I was clicked!")

		Dim quitCommand As New Command With { .MenuText = "Quit", .Shortcut = Application.Instance.CommonModifier Or Keys.Q }
		AddHandler quitCommand.Executed, Sub() Application.Instance.Quit()

		Dim aboutCommand As New Command With { .MenuText = "About..." }
		AddHandler aboutCommand.Executed, Sub()
			Dim about As New AboutDialog()
			about.ShowDialog(Me)
		End Sub

		' create menu
		Dim menuBar = New MenuBar() With {
			.QuitItem = quitCommand,
			.AboutItem = aboutCommand
        }

		' File submenu
		Dim fileMenu As New ButtonMenuItem With { .Text = "&File" }
		fileMenu.Items.Add(clickMe)
		menuBar.Items.Add(fileMenu)

		'Dim editMenu As New ButtonMenuItem With { .Text = "&Edit" }
		'editMenu.Items.Add([command/item])

		'Dim viewMenu as New ButtonMenuItem With { .Text = "&View" }
		'viewMenu.Items.Add([command/item])

		' application (OS X) or file menu (others)
		Dim preferencesMenu As New ButtonMenuItem With { .Text = "&Preferences..." }
		menuBar.ApplicationItems.Add(preferencesMenu)

		Me.Menu = menuBar

		' Create toolbar
		Dim toolBar As New ToolBar()
		toolBar.Items.Add(clickMe)
		Me.ToolBar = toolBar
#End If
#End If
	End Sub

End Class
#End If